components/books-data.js

```diff
@@ -0,0 +1,115 @@
+import { LitElement, html } from "lit-element";
+import styles from "../styles";
+import { BookSearch } from "./books-search";
+
+export class BookData extends LitElement {
+  static get properties() {
+    return {
+      data: [],
+      books: [],
+      allBooks: { type: Array },
+      allRestore: { type: Array }
+    };
+  }
+
+  constructor() {
+    super();
+    this.allBooks = [];
+    this.allRestore=[];
+    this.cargaLibros();
+    this.books = [];
+    this.addEventListener("ApiData", (e) => {
+      this._dataFormat(e.detail.data);
+    });
+    this.dataFromSearch();
+  }
+
+  static get styles() {
+    return [styles];
+  }
+
+  async cargaLibros() {
+    fetch("../books.json").then((respuesta) => respuesta.json());
+  }
+
+  firstUpdated() {
+    this.getData();
+  }
+
+  _sendData(data) {
+    this.dispatchEvent(
+      new CustomEvent("ApiData", {
+        detail: { data },
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  getData() {
+    fetch("../books.json")
+      .then((response) => {
+        if (response.ok) return response.json();
+        return Promise.reject(response);
+      })
```

Comment:
if response has an error, it should throw an exception, you can omit this part and just return the json
```suggestion
      .then((response) => response.json())
```